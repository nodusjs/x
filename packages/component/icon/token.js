import { css } from "@nodusjs/std/dom";

/**
 * @function token
 *
 * @returns {CSSStyleSheet} A folha de estilo contendo todos os tokens de tamanho do ícone.
 *
 * @description
 * Gera uma folha de estilo (`CSSStyleSheet`) que define os Design Tokens
 * para o componente `x-icon`. Estes tokens são variáveis CSS que controlam
 * o tamanho da fonte do ícone para diferentes valores da propriedade `size`.
 *
 * A estrutura dos tokens segue o padrão:
 * `--icon-font-size-[size]`
 *
 * - `size`: O token de tamanho (ex: `sm`, `md`, `lg`, `xl`).
 */
export const token = () =>
  css`
    :host {
      --icon-font-size-sm: 20px;
      --icon-font-size-md: 24px;
      --icon-font-size-lg: 28px;
      --icon-font-size-xl: 32px;
    }
  `;
