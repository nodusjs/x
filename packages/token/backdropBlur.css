:root {
  /**
   * Backdrop blur SM
   * Desfoque leve de fundo para profundidade sutil em sobreposições.
   * Exemplo: pequenas cards ou menus.
   */
  --backdrop-blur-sm: blur(4px);

  /**
   * Backdrop blur MD
   * Desfoque moderado de fundo para elevação média em componentes.
   * Exemplo: painéis de diálogo leves.
   */
  --backdrop-blur-md: blur(8px);

  /**
   * Backdrop blur LG
   * Desfoque intenso de fundo para destacar modais ou overlays grandes.
   */
  --backdrop-blur-lg: blur(16px);

  /**
   * Backdrop blur XL
   * Desfoque pesado de fundo para camadas de fundo fortemente desfocadas
   * em contextos de foco visual.
   */
  --backdrop-blur-xl: blur(24px);
}
