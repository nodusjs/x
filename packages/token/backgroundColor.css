:root {
  /**
   * Fundo primário
   * Cor de fundo primária (branco) usada em todos os layouts e componentes.
   */
  --bg-primary: light-dark(var(--color-base-white), var(--color-gray-950));

  /**
   * Fundo primário alternativo
   * Cor de fundo primária alternativa (branco) que troca para bg-secondary no modo escuro.
   */
  --bg-primary_alt: light-dark(var(--color-base-white), var(--color-gray-900));

  /**
   * Fundo primário (hover)
   * Cor de fundo para estado hover em componentes com fundo branco (ex. itens de dropdown).
   */
  --bg-primary_hover: light-dark(var(--color-gray-50), var(--color-gray-800));

  /**
   * Fundo primário sólido
   * Cor de fundo escura primária usada em layouts e componentes; no modo escuro alterna para bg-secondary.
   * Útil em tooltips e tooltips de editor de texto.
   */
  --bg-primary-solid: light-dark(var(--color-gray-950), var(--color-gray-900));

  /**
   * Fundo secundário
   * Cor de fundo secundária usada para criar contraste contra fundos brancos (ex. seções de site).
   */
  --bg-secondary: light-dark(var(--color-gray-50), var(--color-gray-900));

  /**
   * Fundo secundário alternativo
   * Cor de fundo secundária alternativa que troca para bg-primary no modo escuro.
   * Útil em abas horizontais de borda.
   */
  --bg-secondary_alt: light-dark(var(--color-gray-50), var(--color-gray-950));

  /**
   * Fundo secundário (hover)
   * Cor de fundo para estado hover em componentes com fundo cinza-50 (ex. navegação, date-pickers).
   */
  --bg-secondary_hover: light-dark(var(--color-gray-100), var(--color-gray-800));

  /**
   * Fundo secundário sutil
   * Variante mais clara e sutil em modo claro, útil em banners.
   */
  --bg-secondary_subtle: light-dark(var(--color-gray-25), var(--color-gray-900));

  /**
   * Fundo secundário sólido
   * Cor de fundo escura secundária usada em layouts e componentes;
   * útil em ícones em destaque.
   */
  --bg-secondary-solid: light-dark(var(--color-gray-600), var(--color-gray-600));

  /**
   * Fundo terciário
   * Cor de fundo terciária usada para criar contraste contra fundos claros (ex. toggles).
   */
  --bg-tertiary: light-dark(var(--color-gray-100), var(--color-gray-800));

  /**
   * Fundo quaternário
   * Cor de fundo quaternária usada para criar contraste contra fundos claros (ex. sliders, progress bars).
   */
  --bg-quaternary: light-dark(var(--color-gray-200), var(--color-gray-700));

  /**
   * Fundo ativo
   * Cor de fundo padrão para estado ativo em componentes (ex. item selecionado em dropdown).
   */
  --bg-active: light-dark(var(--color-gray-50), var(--color-gray-800));

  /**
   * Fundo desabilitado
   * Cor de fundo padrão para componentes desabilitados (ex. botões primários e toggles).
   */
  --bg-disabled: light-dark(var(--color-gray-100), var(--color-gray-800));

  /**
   * Fundo desabilitado sutil
   * Variante mais sutil para componentes desabilitados (ex. campos de formulário, checkboxes).
   */
  --bg-disabled_subtle: light-dark(var(--color-gray-50), var(--color-gray-900));

  /**
   * Fundo overlay
   * Cor de fundo para overlays, útil em modals.
   */
  --bg-overlay: light-dark(var(--color-gray-950), var(--color-gray-800));

  /**
   * Fundo da marca primária
   * Cor de fundo primária da marca, útil em componentes como ícones de confirmação.
   */
  --bg-brand-primary: light-dark(var(--color-brand-50), var(--color-brand-500));

  /**
   * Fundo da marca primária alternativo
   * Cor de fundo da marca que troca para bg-secondary no modo escuro (ex. abas horizontais ativas).
   */
  --bg-brand-primary_alt: light-dark(
    var(--color-brand-50),
    var(--color-gray-800)
  );

  /**
   * Fundo da marca secundária
   * Cor de fundo secundária da marca, útil em componentes como ícones em destaque.
   */
  --bg-brand-secondary: light-dark(
    var(--color-brand-100),
    var(--color-brand-600)
  );

  /**
   * Fundo sólido da marca
   * Cor sólida de fundo da marca usada em toggles e mensagens.
   */
  --bg-brand-solid: light-dark(var(--color-brand-600), var(--color-brand-600));

  /**
   * Fundo sólido da marca (hover)
   * Cor sólida de fundo da marca em estado hover, útil em toggles.
   */
  --bg-brand-solid_hover: light-dark(
    var(--color-brand-700),
    var(--color-brand-500)
  );

  /**
   * Fundo de seção da marca
   * Cor de fundo padrão para seções de site (ex. CTAs, testimonials); no modo escuro alterna para bg-secondary.
   */
  --bg-brand-section: light-dark(var(--color-brand-800), var(--color-gray-800));

  /**
   * Fundo de seção da marca sutil
   * Variante sutil para seções de FAQ; no modo escuro alterna para bg-primary.
   */
  --bg-brand-section_subtle: light-dark(
    var(--color-brand-700),
    var(--color-gray-950)
  );

  /**
   * Fundo de erro primário
   * Cor semântica primária de erro para componentes (ex. botões).
   */
  --bg-error-primary: light-dark(var(--color-error-50), var(--color-error-950));

  /**
   * Fundo de erro secundário
   * Cor semântica secundária de erro, útil em ícones em destaque.
   */
  --bg-error-secondary: light-dark(
    var(--color-error-100),
    var(--color-error-600)
  );

  /**
   * Fundo sólido de erro
   * Cor sólida de fundo de erro usada em botões, ícones em destaque e metric items.
   */
  --bg-error-solid: light-dark(var(--color-error-600), var(--color-error-600));

  /**
   * Fundo sólido de erro (hover)
   * Cor sólida de fundo de erro em estado hover, útil em botões.
   */
  --bg-error-solid_hover: light-dark(
    var(--color-error-700),
    var(--color-error-500)
  );

  /**
   * Fundo de aviso primário
   * Cor semântica primária de aviso para componentes.
   */
  --bg-warning-primary: light-dark(
    var(--color-warning-50),
    var(--color-warning-950)
  );

  /**
   * Fundo de aviso secundário
   * Cor semântica secundária de aviso, útil em ícones destacados.
   */
  --bg-warning-secondary: light-dark(
    var(--color-warning-100),
    var(--color-warning-600)
  );

  /**
   * Fundo sólido de aviso
   * Cor sólida de fundo de aviso usada em ícones em destaque.
   */
  --bg-warning-solid: light-dark(
    var(--color-warning-600),
    var(--color-warning-600)
  );

  /**
   * Fundo de sucesso primário
   * Cor semântica primária de sucesso para componentes.
   */
  --bg-success-primary: light-dark(
    var(--color-success-50),
    var(--color-success-950)
  );

  /**
   * Fundo de sucesso secundário
   * Cor semântica secundária de sucesso, útil em ícones em destaque.
   */
  --bg-success-secondary: light-dark(
    var(--color-success-100),
    var(--color-success-600)
  );

  /**
   * Fundo sólido de sucesso
   * Cor sólida de fundo de sucesso usada em ícones em destaque e metric items.
   */
  --bg-success-solid: light-dark(
    var(--color-success-600),
    var(--color-success-600)
  );
}
