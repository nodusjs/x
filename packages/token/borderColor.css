:root {
	/**
	 * Borda primária
	 * Bordas de alto contraste usadas em componentes como campos de entrada,
	 * grupos de botões e checkboxes.
	 */
	--border-primary: light-dark(var(--color-gray-300), var(--color-gray-700));

	/**
	 * Borda secundária
	 * Bordas de contraste médio, padrão para a maioria dos componentes
	 * (file uploaders, cards e divisores de conteúdo).
	 */
	--border-secondary: light-dark(var(--color-gray-200), var(--color-gray-800));

	/**
	 * Borda secundária alternativa
	 * Variante com transparência alfa (8%) para menus flutuantes
	 * como dropdowns e notificações, criando uma linha inferior mais nítida.
	 */
	--border-secondary_alt: light-dark(rgba(0, 0, 0, 0.08), var(--color-gray-800));

	/**
	 * Borda terciária
	 * Bordas de baixo contraste para divisores muito sutis,
	 * como eixos de gráficos de linha e barras.
	 */
	--border-tertiary: light-dark(var(--color-gray-100), var(--color-gray-800));

	/**
	 * Borda desabilitada
	 * Cor padrão para estados desabilitados em componentes
	 * como campos de entrada e checkboxes.
	 */
	--border-disabled: light-dark(var(--color-gray-300), var(--color-gray-700));

	/**
	 * Borda desabilitada sutil
	 * Variante de contraste ainda menor para bordas desabilitadas,
	 * útil para botões no estado desativado.
	 */
	--border-disabled_subtle: light-dark(
		var(--color-gray-200),
		var(--color-gray-800)
	);

	/**
	 * Borda da marca
	 * Cor de borda padrão da marca, útil para estados ativos
	 * em componentes como campos de entrada.
	 */
	--border-brand: light-dark(var(--color-brand-500), var(--color-brand-400));

	/**
	 * Borda da marca alternativa
	 * Variante que se torna cinza no modo escuro, usada em
	 * componentes estilo “banner” e rodapés da marca.
	 */
	--border-brand_alt: light-dark(var(--color-brand-600), var(--color-gray-700));

	/**
	 * Borda de erro
	 * Cor semântica padrão para bordas em estado de erro,
	 * usada em componentes como campos de formulário e uploaders.
	 */
	--border-error: light-dark(var(--color-error-500), var(--color-error-400));

	/**
	 * Borda de erro sutil
	 * Variante de contraste reduzido para bordas semânticas de erro,
	 * útil em campos de formulário com estado de erro menos severo.
	 */
	--border-error_subtle: light-dark(
		var(--color-error-300),
		var(--color-error-500)
	);
}
