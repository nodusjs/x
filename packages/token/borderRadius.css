:root {
  /**
   * radius-none
   * Sem arredondamento; raio zero.
   */
  --radius-none: 0px;

  /**
   * radius-xxs
   * Raio de 2px para cantos levemente arredondados.
   */
  --radius-xxs: 2px;

  /**
   * radius-xs
   * Raio de 4px para cantos suavemente arredondados.
   */
  --radius-xs: 4px;

  /**
   * radius-sm
   * Raio de 6px para cantos arredondados padrão.
   */
  --radius-sm: 6px;

  /**
   * radius-md
   * Raio de 8px para cantos arredondados moderados.
   */
  --radius-md: 8px;

  /**
   * radius-lg
   * Raio de 10px para cantos arredondados grandes.
   */
  --radius-lg: 10px;

  /**
   * radius-xl
   * Raio de 12px para cantos arredondados extra grandes.
   */
  --radius-xl: 12px;

  /**
   * radius-2xl
   * Raio de 16px para cantos extra extra grandes.
   */
  --radius-2xl: 16px;

  /**
   * radius-3xl
   * Raio de 20px para cantos ainda mais arredondados.
   */
  --radius-3xl: 20px;

  /**
   * radius-4xl
   * Raio de 24px para bordas fortemente arredondadas.
   */
  --radius-4xl: 24px;

  /**
   * radius-full
   * Raio máximo (9999px) para perfis totalmente circulares.
   */
  --radius-full: 9999px;
}
