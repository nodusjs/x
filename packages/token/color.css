:root {
  /**
   * Base
   *
   * Estas são as cores base (branco e preto) para você alternar rapidamente,
   * caso precise.
   */
  --color-base-white: #ffffff;
  --color-base-black: #000000;
  --color-base-transparent: #ffffff 0%;

  /**
   * Cinza (modo claro)
   *
   * Cinza é uma cor neutra e é a base do sistema de cores.
   * Quase tudo em design de interface — texto, campos de formulário,
   * fundos, divisores — normalmente é cinza.
   */
  @media (prefers-color-scheme: light) {
    --color-gray-25: #fdfdfd;
    --color-gray-50: #fafafa;
    --color-gray-100: #f5f5f5;
    --color-gray-200: #e9eaeb;
    --color-gray-300: #d5d7da;
    --color-gray-400: #a4a7ae;
    --color-gray-500: #717680;
    --color-gray-600: #535862;
    --color-gray-700: #414651;
    --color-gray-800: #252b37;
    --color-gray-900: #181d27;
    --color-gray-950: #0a0d12;
  }

  /**
   * Cinza (modo escuro)
   *
   * Esta é uma paleta de cinza separada usada especificamente para o modo escuro.
   * Essa cor foi dessaturada e otimizada para funcionar bem com variáveis do modo escuro.
   */
  @media (prefers-color-scheme: dark) {
    --color-gray-25: #fafafa;
    --color-gray-50: #f7f7f7;
    --color-gray-100: #f0f0f1;
    --color-gray-200: #ececed;
    --color-gray-300: #cecfd2;
    --color-gray-400: #94979c;
    --color-gray-500: #85888e;
    --color-gray-600: #61656c;
    --color-gray-700: #373a41;
    --color-gray-800: #22262f;
    --color-gray-900: #13161b;
    --color-gray-950: #0c0e12;
  }

  /**
   * Marca
   *
   * A cor da marca é sua cor “primária” e é usada em todos os elementos
   * interativos — botões, links, inputs etc. — podendo definir a sensação
   * geral e evocar emoções.
   */
  --color-brand-25: #fcfaff;
  --color-brand-50: #f9f5ff;
  --color-brand-100: #f4ebff;
  --color-brand-200: #e9d7fe;
  --color-brand-300: #d6bbfb;
  --color-brand-400: #b692f6;
  --color-brand-500: #9e77ed;
  --color-brand-600: #7f56d9;
  --color-brand-700: #6941c6;
  --color-brand-800: #53389e;
  --color-brand-900: #42307d;
  --color-brand-950: #2c1c5f;

  /**
   * Erro
   *
   * Cores de erro são usadas em estados de erro e em ações “destrutivas”.
   * Elas comunicam uma ação negativa, como remover um usuário da equipe.
   */
  --color-error-25: #fffbfa;
  --color-error-50: #fef3f2;
  --color-error-100: #fee4e2;
  --color-error-200: #fecdca;
  --color-error-300: #fda29b;
  --color-error-400: #f97066;
  --color-error-500: #f04438;
  --color-error-600: #d92d20;
  --color-error-700: #b42318;
  --color-error-800: #912018;
  --color-error-900: #7a271a;
  --color-error-950: #55160c;

  /**
   * Aviso
   *
   * Cores de aviso podem indicar que uma ação é potencialmente destrutiva
   * ou “em espera”. São comumente usadas em confirmações para chamar
   * a atenção do usuário.
   */
  --color-warning-25: #fffcf5;
  --color-warning-50: #fffaeb;
  --color-warning-100: #fef0c7;
  --color-warning-200: #fedf89;
  --color-warning-300: #fec84b;
  --color-warning-400: #fdb022;
  --color-warning-500: #f79009;
  --color-warning-600: #dc6803;
  --color-warning-700: #b54708;
  --color-warning-800: #93370d;
  --color-warning-900: #7a2e0e;
  --color-warning-950: #4e1d09;

  /**
   * Sucesso
   *
   * Cores de sucesso comunicam ações positivas, tendências favoráveis
   * ou confirmações bem-sucedidas. Útil diferenciar do verde primário.
   */
  --color-success-25: #f6fef9;
  --color-success-50: #ecfdf3;
  --color-success-100: #dcfae6;
  --color-success-200: #abfec6;
  --color-success-300: #75e0a7;
  --color-success-400: #47cd89;
  --color-success-500: #17b26a;
  --color-success-600: #079455;
  --color-success-700: #067647;
  --color-success-800: #085d3a;
  --color-success-900: #074d31;
  --color-success-950: #053321;
}
