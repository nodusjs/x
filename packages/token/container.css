:root {
  /**
   * container-padding-mobile
   * Espaçamento interno padrão para containers em dispositivos móveis.
   * Padding: 16px
   */
  --container-padding-mobile: 16px;

  /**
   * container-padding-desktop
   * Espaçamento interno padrão para containers em desktop.
   * Padding: 32px
   */
  --container-padding-desktop: 32px;

  /**
   * container-max-width-desktop
   * Largura máxima dos containers em desktop para centralizar o conteúdo.
   * Max-width: 1280px
   */
  --container-max-width-desktop: 1280px;
}
