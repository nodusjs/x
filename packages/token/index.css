@import url("./backdropBlur.css");
@import url("./backgroundColor.css");
@import url("./borderColor.css");
@import url("./borderRadius.css");
@import url("./color.css");
@import url("./container.css");
@import url("./focus.css");
@import url("./foregroundColor.css");
@import url("./paragraph.css");
@import url("./shadow.css");
@import url("./spacing.css");
@import url("./textColor.css");
@import url("./typography.css");
@import url("./width.css");
