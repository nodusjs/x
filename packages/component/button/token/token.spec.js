import { describe, it, expect } from 'vitest'
import { token } from './token.js'

describe('token', () => {
  it('gera o bloco de tokens CSS esperado', () => {
    const { cssRules: [{ cssText }]} = token()
    expect(cssText).toMatchSnapshot()
  })
})
