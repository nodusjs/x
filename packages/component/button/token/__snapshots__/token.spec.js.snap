// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`token > gera o bloco de tokens CSS esperado 1`] = `":host { --button-background-color-brand-solid: var(--bg-brand-solid); --button-background-color-brand-outlined: var(--bg-primary); --button-background-color-brand-ghost: transparent !important; --button-background-color-brand-link: transparent !important; --button-background-color-brand-solid_hover: var(--bg-brand-solid_hover); --button-background-color-brand-outlined_hover: var(--bg-primary_hover); --button-background-color-brand-ghost_hover: var(--bg-primary_hover); --button-background-color-brand-link_hover: transparent !important; --button-border-brand-solid: 1px solid var(--bg-brand-solid); --button-border-brand-outlined: 1px solid var(--border-disabled_subtle); --button-border-brand-ghost: none !important; --button-border-brand-link: nonde !important; --button-color-brand-solid: var(--color-base-white); --button-color-brand-outlined: var(--text-secondary); --button-color-brand-ghost: var(--text-tertiary); --button-color-brand-link: var(--text-brand-secondary); --button-color-brand-solid_hover: var(--color-base-white); --button-color-brand-outlined_hover: var(--text-secondary_hover); --button-color-brand-ghost_hover: var(--text-tertiary_hover); --button-color-brand-link_hover: var(--text-brand-secondary_hover); --button-background-color-error-solid: var(--bg-error-solid); --button-background-color-error-outlined: var(--bg-primary); --button-background-color-error-ghost: transparent !important; --button-background-color-error-link: transparent !important; --button-background-color-error-solid_hover: var(--bg-error-solid_hover); --button-background-color-error-outlined_hover: var(--bg-error-primary); --button-background-color-error-ghost_hover: var(--bg-error-primary); --button-background-color-error-link_hover: transparent !important; --button-border-error-solid: 1px solid var(--bg-error-solid_hover); --button-border-error-outlined: 1px solid var(--border-error_subtle); --button-border-error-ghost: none !important; --button-border-error-link: nonde !important; --button-color-error-solid: var(--color-base-white); --button-color-error-outlined: var(--text-error-primary); --button-color-error-ghost: var(--text-error-primary); --button-color-error-link: var(--text-error-primary); --button-color-error-solid_hover: var(--color-base-white); --button-color-error-outlined_hover: var(--text-error-primary_hover); --button-color-error-ghost_hover: var(--text-error-primary_hover); --button-color-error-link_hover: var(--text-error-primary_hover); --button-font-size-sm: var(--font-size-text-sm); --button-font-size-md: var(--font-size-text-sm); --button-font-size-lg: var(--font-size-text-md); --button-font-size-xl: var(--font-size-text-md); --button-gap-sm: var(--spacing-xs); --button-gap-md: var(--spacing-xs); --button-gap-lg: var(--spacing-sm); --button-gap-xl: var(--spacing-sm); --button-line-height-sm: var(--line-height-text-sm); --button-line-height-md: var(--line-height-text-sm); --button-line-height-lg: var(--line-height-text-md); --button-line-height-xl: var(--line-height-text-md); --button-height-sm: 36px; --button-height-md: 40px; --button-height-lg: 44px; --button-height-xl: 48px; --button-width-sm: 36px; --button-width-md: 40px; --button-width-lg: 44px; --button-width-xl: 48px; --button-box-shadow-brand: var(--focus-ring); --button-box-shadow-error: var(--focus-ring-error); }"`;
